src/classes/Query.cls

```diff
@@ -1639,6 +1658,9 @@ public class Query {
             query += ' OFFSET ' + resultOffset;
         }
 
+        if(!optionalClause.isEmpty()){
```

Comment:
Seems that `isEmpty()` is not a method of String type. It would fail in compilation.